vllm-proxy/tests/signature_verification_example.py

```diff
@@ -0,0 +1,304 @@
+
+    print(f"Chat ID: {chat_id}")
+    print(f"Calculated response hash: {response_hash}")
+
+    return chat_id, response_hash
+
+
+def example_streaming_without_hash(base_url, auth_token, request_body_str):
+    """Example 2: Streaming request without hash header"""
+    print("\n=== Example 2: Streaming request without hash (let server calculate) ===")
+
+    response = requests.post(
+        f"{base_url}/v1/chat/completions",
+        headers={
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {auth_token}",
+        },
+        data=request_body_str,
+        stream=True,
+    )
+
+    result = process_response_stream(response, calculate_hash=False)
+    chat_id = result["chat_id"]
+    response_text = result["response_text"]
+    response_hash = calculate_response_hash(response_text)
+
+    print(f"Chat ID: {chat_id}")
+
+    return chat_id, response_hash
+
+
+def example_non_streaming_with_hash(
+    base_url, auth_token, request_body_str, calculated_hash
+):
+    """Example 3: Non-streaming request with pre-calculated hash"""
+    print("\n=== Example 3: Non-streaming request with pre-calculated hash ===")
+    print(f"Calculated request hash: {calculated_hash}")
+
+    response = requests.post(
+        f"{base_url}/v1/chat/completions",
+        headers={
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {auth_token}",
+            "X-Request-Hash": calculated_hash,
+        },
+        data=request_body_str,
+    )
+
+    result = process_response_non_stream(response)
+    chat_id = result["chat_id"]
+    response_hash = result["response_hash"]
+
+    print(f"Chat ID: {chat_id}")
+    print(f"Calculated response hash: {response_hash}")
+
+    return chat_id, response_hash
+
+
+def example_non_streaming_without_hash(base_url, auth_token, request_body_str):
+    """Example 4: Non-streaming request without hash header"""
+    print("\n=== Example 4: Non-streaming request without hash ===")
+
+    response = requests.post(
+        f"{base_url}/v1/chat/completions",
+        headers={
+            "Content-Type": "application/json",
+            "Authorization": f"Bearer {auth_token}",
+        },
+        data=request_body_str,
+    )
+
+    result = process_response_non_stream(response)
+    chat_id = result["chat_id"]
+    response_hash = result["response_hash"]
+
+    print(f"Chat ID: {chat_id}")
+    print(f"Calculated response hash: {response_hash}")
+
+    return chat_id, response_hash
+
+
+def main():
+    base_url = os.getenv("VLLM_BASE_URL")
+    auth_token = os.getenv("AUTH_TOKEN")
+
+    # Streaming request payload
+    stream_request_body = {
+        "model": "meta/llama-3.3-70b-instruct",
+        "messages": [{"role": "user", "content": "Hello, how are you?"}],
+        "stream": True,
+        "max_tokens": 1,
+    }
+    stream_request_str = json.dumps(
+        stream_request_body, separators=(",", ":"), sort_keys=True
```

Comment:
Are the separators and sort_keys options necessary? If no, I suggest to skip it. My understanding is that we just hash the body anyway.